src/System.Text.Json/src/System/Text/Json/JsonUtf8Reader.cs

```diff
@@ -0,0 +1,1274 @@
+        /// always return false. For input data within a ReadOnlySequence&lt;byte&gt;, this
+        /// will only return true if the token straddles more than a single segment and
+        /// hence couldn't be represented as a span.
+        /// </summary>
+        public bool IsValueMultiSegment { get; private set; }
+
+        /// <summary>
+        /// Gets the value of the last processed token as a ReadOnlySpan&lt;byte&gt; slice
+        /// of the input payload. If the JSON is provided within a ReadOnlySequence&lt;byte&gt;
+        /// and the slice that represents the token value fits in a single segment, then
+        /// ValueSpan will contain the sliced value since it can be represented as a span.
+        /// </summary>
+        public ReadOnlySpan<byte> ValueSpan { get; private set; }
+
+        /// <summary>
+        /// Gets the value of the last processed token as a ReadOnlySequence&lt;byte&gt; slice
+        /// of the input payload. If the JSON is provided within a ReadOnlySpan&lt;byte&gt;
+        /// this will always return an empty sequence. Also, if the JSON is provided within a
+        /// ReadOnlySequence&lt;byte&gt; and the slice that represents the token value fits in
+        /// a single segment, then ValueSpan will contain the sliced value since it can be
+        /// represented as a span.
+        /// </summary>
+        public ReadOnlySequence<byte> ValueSequence { get; private set; }
+
+        /// <summary>
+        /// Returns the total amount of bytes consumed by the <see cref="JsonUtf8Reader"/> so far
+        /// for the given UTF-8 encoded input text.
+        /// </summary>
+        public long BytesConsumed => _totalConsumed + _consumed;
+
+        // Depth tracks the recursive depth of the nested objects / arrays within the JSON data.
+        /// <summary>
+        /// Tracks the recursive depth of the nested objects / arrays within the JSON text
+        /// processed so far. This provides the depth of the current token.
+        /// </summary>
+        public int CurrentDepth => _currentDepth;
+
+        /// <summary>
+        /// Gets the type of the last processed JSON token in the UTF-8 encoded JSON text.
+        /// </summary>
+        public JsonTokenType TokenType => _tokenType;
+
+        /// <summary>
+        /// Returns the current <see cref="SequencePosition"/> within the provided UTF-8 encoded
+        /// input ReadOnlySequence&lt;byte&gt;. If the <see cref="JsonUtf8Reader"/> was constructed
+        /// with a ReadOnlySpan&lt;byte&gt; instead, this will always return a default <see cref="SequencePosition"/>.
+        /// </summary>
+        public SequencePosition Position
+        {
+            get
+            {
+                if (_currentPosition.GetObject() == null)
+                    return default;
+
+                return _sequence.GetPosition(BytesConsumed);
+                // TODO: This fails - return _data.GetPosition(_consumed - _leftOverLength, _currentPosition);
```

Comment:
What's the TODO here? Does it need an issue?

Reply:
Yes. I am going to re-validate and file an issue.